File: typescript/src/workshop/m3/s07_srp/step3/DailyReport.ts
import type { Sale } from '../Sale.js';
import { AccountingSection } from './AccountingSection.js';
import { MarketingSection } from './MarketingSection.js';

/**
 * Krok 3 (rozwiązanie): Extract Class - każda sekcja w klasie swojego aktora.
 * DailyReport tylko składa dokument (koordynuje, nie zna polityk). Zmiana definicji
 * hitu dotyka wyłącznie `MarketingSection`.
 */
export class DailyReport {
  private readonly accounting = new AccountingSection();
  private readonly marketing = new MarketingSection();

  render(sales: readonly Sale[]): string {
    return this.accounting.render(sales) + this.marketing.render(sales);
  }
}

File: csharp/src/Training.Workshop/M3/S07Srp/Step3/DailyReport.cs
namespace Training.Workshop.M3.S07Srp.Step3;

/// <summary>
/// Krok 3 (rozwiązanie): Extract Class - każda sekcja w klasie swojego aktora.
/// DailyReport tylko składa dokument (koordynuje, nie zna polityk). Zmiana definicji
/// hitu dotyka wyłącznie <see cref="MarketingSection"/>.
/// </summary>
public sealed class DailyReport
{
    private readonly AccountingSection _accounting = new();
    private readonly MarketingSection _marketing = new();

    public string Render(IReadOnlyList<Sale> sales)
    {
        return _accounting.Render(sales) + _marketing.Render(sales);
    }
}
